prover_rust/src/main.rs

```diff
@@ -0,0 +1,141 @@
+mod config;
+mod coordinator_client;
+mod geth_client;
+mod key_signer;
+mod prover;
+mod task_cache;
+mod types;
+mod utils_log;
+mod version;
+mod zk_circuits_handler;
+
+use anyhow::{Context, Result};
+use config::Config;
+use coordinator_client::listener::Listener;
+use log;
+use prover::Prover;
+use std::rc::Rc;
+use task_cache::TaskCache;
+use types::TaskWrapper;
+
+struct ClearCacheCoordinatorListener {
```

Comment:
Why don't put it together with task_cache?

Reply:
done